### PR TITLE
campaigns: Extend campaign spec schema with `workspaces`

### PR DESCRIPTION
**Pull request by mrnugget**
This PR contains the schema changes necessary for the automatic workspace discovery implemented in https://github.com/sourcegraph/src-cli/pull/442.

### CODE REVIEW

**Comment by mrnugget** [comment]
Requesting review, because I'd love to merge this independently of documentation.

File: schema/campaign_spec.schema.json [hunk 1]
```diff
@@ -56,6 +56,28 @@
         ]
       }
     },
+    "workspaces": {
+      "type": "array",
+      "description": "Different workspace configurations",
```
Comment (eseliger) [inline schema/campaign_spec.schema.json:61]:
```suggestion
      "description": "Individual workspace configurations",
```

maybe? I didn't really understand what is _different_ here

File: schema/campaign_spec.schema.json [hunk 2]
```diff
@@ -56,6 +56,28 @@
...
+        "additionalProperties": false,
+        "required": ["rootAtLocationOf"],
+        "properties": {
+          "rootAtLocationOf": {
+            "type": "string",
+            "description": "The name of the file that sits at the root of the desired workspace"
+          },
+          "in": {
+            "type": "string",
+            "description": "The repositories in which to apply the workspace configuration",
```
Comment (eseliger) [inline schema/campaign_spec.schema.json:75]:
```suggestion
            "description": "The repositories in which to apply the workspace configuration. Supports globbing",
```

**Review by eseliger — APPROVED** [review]
just some ideas, no blockers

File: schema/campaign_spec.schema.json (hunk 1 above)
Comment (mrnugget) [reply schema/campaign_spec.schema.json:61]: Yes, correct. Sorry, didn't put a lot of effort into the descriptions this time.

**Pull request merged**